Docs sidecar analysis (#15669)
kyma-project/kyma: docs/04-operation-guides/operations/smsh-01-istio-enable-sidecar-injection.md

Read the [Istio documentation](https://istio.io/docs/setup/kubernetes/additional-setup/sidecar-injection/) to learn more about sidecar proxy injection and consider [benefits of having the sidecar container inside your application pod](../../01-overview/main-areas/service-mesh/smsh-03-istio-sidecars-in-kyma.md).

If there are issues with the Istio sidecar, you can check whether there is an [issue with the sidecar injection](../troubleshooting/service-mesh/smsh-03-istio-no-sidecar.md) or a [mismatching Istio version](../troubleshooting/service-mesh/smsh-04-istio-sidecar-version.md).

## Check whether your workloads have automatic Istio sidecar injection enabled

> You can easily check whether your workloads have automatic Istio sidecar injection enabled by running [this script](../assets/sidecar-analysis.sh). You can either pass a namespace parameter to the script or run with no parameter.

Reviewer: ```suggestion
You can easily check whether your workloads have automatic Istio sidecar injection enabled by running [this script](../assets/sidecar-analysis.sh). You can either pass a **namespace** parameter to the script or run it with no parameter.
```

> If no parameter is passed execution output will contain pods in all namespaces that don't have automatic Istio sidecar injection enabled, whereas passing the parameter will result in analysis of only the given namespace.

Reviewer: ```suggestion
If no parameter is passed, the execution output will contain Pods from all Namespaces that don't have automatic Istio sidecar injection enabled, whereas passing the parameter results in the analysis of only the given  Namespace.
```

> The script outputs the information in `{namespace}/{pod}` if run for all namespaces and in `{pod}` form for specific namespace.

Reviewer: ```suggestion
The script outputs the information in `{namespace}/{pod}` if run for all Namespaces and in `{pod}` form for a specific Namespace.
```

> * Running the script

Reviewer: ```suggestion
* Run the script
```
